### PR TITLE
feat: add project setup with kelvin dependencies

### PR DESCRIPTION
**Pull request by GoncaloCamaz**
- [Kelvin Angular SDK](https://www.npmjs.com/package/@kelvininc/angular-client-sdk)
- [Kelvin UI Components](https://www.npmjs.com/package/@kelvininc/ui-components)
- [Angular Keycloak](https://www.npmjs.com/package/@kelvininc/angular-client-sdk)

### CODE REVIEW

**Review by GoncaloCamaz — COMMENTED** [review]
AuthGuard needs changes on the method isAccessAllowed since the second if can be an also or discarded 

```
isAccessAllowed(): Promise<boolean> {
		return new Promise(async (resolve, _) => {
			if (!this.authenticated) {
				this.sessionService.clearSession();
				this.keycloakAngular.login();

				return resolve(false);
			}
			else
			{
				return resolve(true);
			}
		});
	}
```

File: src/app/pages/nodes-list/nodes-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
+import { Component, OnInit } from '@angular/core';
+import { ACPItem, KvPaginator, ACPService } from '@kelvininc/angular-client-sdk';
+import { Subscription } from 'rxjs';
+import { NUMBER_OF_NODES } from './nodes-list.config';
+import { NodesListState } from './nodes-list.types';
+
+const initialState: NodesListState = {
```
Comment (pedrorebelocosta) [inline src/app/pages/nodes-list/nodes-list.component.ts:7]: you can move this inside the component itself ;) you won't be able to use the const keyword  (this applies for the other ones)

**Review by pedrorebelocosta — CHANGES_REQUESTED** [review]

File: src/app/app.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    const fixture = TestBed.createComponent(AppComponent);
+    const app = fixture.componentInstance;
+    expect("").toEqual('kelvin-angular-sample-app');
+  });
+
+  it('should render title', () => {
+    const fixture = TestBed.createComponent(AppComponent);
+    fixture.detectChanges();
+    const compiled = fixture.nativeElement as HTMLElement;
+    expect(compiled.querySelector('.content span')?.textContent).toContain('kelvin-angular-sample-app app is running!');
```
[the remark this answers is not in the document]
Comment (GoncaloCamaz) [reply src/app/app.component.spec.ts:33]: In this case I will change the test to another html component check

File: src/app/app.component.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	selector: 'app-root',
+	templateUrl: './app.component.html'
+})
+export class AppComponent {
+	constructor(
+		private appService: ApplicationService,
+		private sessionService: SessionService,
+		private router: Router,
+	) {
+		if (!environment.production) {
```
[the remark this answers is not in the document]
Comment (GoncaloCamaz) [reply src/app/app.component.ts:19]: Will remove

File: src/app/pages/assets-list/assets-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  }
+}
+
+@Component({
+  selector: 'app-assets-list',
+  templateUrl: './assets-list.component.html',
+  styleUrls: ['./assets-list.component.scss']
+})
+export class AssetsListComponent implements OnInit {
+  assetsState: AssetsListState = initialState
```
[the remark this answers is not in the document]
Comment (GoncaloCamaz) [reply src/app/pages/assets-list/assets-list.component.ts:21]: Will add semicolons on all instance variables

File: src/app/pages/nodes-list/nodes-list.component.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+@Component({
+  selector: 'app-nodes-list',
+  templateUrl: './nodes-list.component.html',
+  styleUrls: ['./nodes-list.component.scss']
+})
+export class NodesListComponent implements OnInit {
+  nodesState: NodesListState = initialState
+  paginator: KvPaginator<ACPItem> | undefined 
+  subscription: Subscription = new Subscription;
```
[the remark this answers is not in the document]
Comment (GoncaloCamaz) [reply src/app/pages/nodes-list/nodes-list.component.ts:23]: It does work but its not necessary, will remove on the next commit

File: src/app/pages/nodes-list/nodes-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+    hasPrevious: false
+  }
+}
+
+@Component({
+  selector: 'app-nodes-list',
+  templateUrl: './nodes-list.component.html',
+  styleUrls: ['./nodes-list.component.scss']
+})
+export class NodesListComponent implements OnInit {
```
[the remark this answers is not in the document]
Comment (GoncaloCamaz) [reply src/app/pages/nodes-list/nodes-list.component.ts:20]: Will correct as well on the next commit

File: src/app/pages/nodes-list/nodes-list.component.ts (hunk 1 above)
Comment (GoncaloCamaz) [reply src/app/pages/nodes-list/nodes-list.component.ts:7]: Will move to the beginning of the component on the next commit

File: src/environments/environment.config.ts [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+		{
+			arch: '64-BIT',
+			os: 'Ubuntu',
+			url: 'https://kelvin-builder.s3.amazonaws.com/0.1.3/kelvin-builder_0.1.3_amd64.deb',
+			version: '0.1.3'
+		}
+	]
+};
+
+const MOCK_METADATA: IMetadataSchema = {
```
[the remark this answers is not in the document]
Comment (ricardogoncalves89) [reply src/environments/environment.config.ts:48]: I check if you really need these configs

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]

File: package.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    "@angular/common": "~13.2.0",
+    "@angular/compiler": "~13.2.0",
+    "@angular/core": "~13.2.0",
+    "@angular/forms": "~13.2.0",
+    "@angular/platform-browser": "~13.2.0",
+    "@angular/platform-browser-dynamic": "~13.2.0",
+    "@angular/router": "~13.2.0",
+    "@kelvininc/angular-client-sdk": "^7.1.0",
+    "@kelvininc/angular-ui-components": "^0.5.1",
+    "@kelvininc/ui-components": "^0.6.0",
```
Comment (ricardogoncalves89) [inline package.json:23]: You don't need this dependency only the `@kelvininc/angular-ui-components`

File: src/app/pages/assets-list/assets-list.component.html [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+			<td>Asset Type</td>
+		</tr>
+		<tr *ngFor="let asset of assetsState.assetsList">
+			<td>{{asset.name}}</td>
+			<td>{{asset.title}}</td>
+			<td>{{asset.assetTypeTitle}}</td>
+		</tr>
+	</tbody>
+</table>
+<div className="paginator-buttons-container">
```
Comment (ricardogoncalves89) [inline src/app/pages/assets-list/assets-list.component.html:15]: `class` not `className` like react 😉

Comment (ricardogoncalves89) [reply src/app/pages/assets-list/assets-list.component.html:15]: you don't have any definition for that css class.

File: src/app/pages/assets-list/assets-list.component.ts [hunk 10]
```diff
@@ -0,0 +1,45 @@
+import { Component, OnInit } from '@angular/core';
+import { AssetItem, AssetService, KvPaginator } from '@kelvininc/angular-client-sdk'
+import { Subscription } from 'rxjs';
+import { NUMBER_OF_ASSETS } from './assets-list.config';
+import { AssetsListState } from './assets-list.types';
+
+@Component({
+	selector: 'app-assets-list',
+	templateUrl: './assets-list.component.html',
+	styleUrls: ['./assets-list.component.scss']
```
Comment (ricardogoncalves89) [inline src/app/pages/assets-list/assets-list.component.ts:10]: If you don't have any css style in the file you can remove it

File: src/app/pages/nodes-list/nodes-list.component.html [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+			<td>Node Status</td>
+		</tr>
+		<tr *ngFor="let node of nodesState.nodesList">
+			<td>{{node.name}}</td>
+			<td>{{node.title}}</td>
+			<td>{{node.status.state}}</td>
+		</tr>
+	</tbody>
+</table>
+<div className="paginator-buttons-container">
```
Comment (ricardogoncalves89) [inline src/app/pages/nodes-list/nodes-list.component.html:15]: same

**Review by ricardogoncalves89 — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (GoncaloCamaz) [reply package.json:23]: Will remove the unnecessary dependency and update to the newer version :)

File: src/app/pages/assets-list/assets-list.component.html (hunk 9 above)
Comment (GoncaloCamaz) [reply src/app/pages/assets-list/assets-list.component.html:15]: Noob mistake :D Will correct

**Comment by nelsonmestevao** [comment]
It is now working with our latest version of '@kelvininc/angular-ui-components` :raised_hands:

**Review by miguelpinto98 — CHANGES_REQUESTED** [review]
It's almost ok. Only some improvements and a security check.

File: src/app/app.module.ts [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+import { SessionService } from './services/session.service';
+import { HomeComponent } from './pages/home/home.component';
+import { NodesListComponent } from './pages/nodes-list/nodes-list.component';
+import { AssetsListComponent } from './pages/assets-list/assets-list.component';
+import { KvUIComponentsModule } from '@kelvininc/angular-ui-components';
+
+@NgModule({
+	declarations: [
+		AppComponent,
+		HomeComponent,
```
[the remark this answers is not in the document]
Comment (nelsonmestevao) [reply src/app/app.module.ts:19]: What do you mean? `HomeModule`?

[the remark this answers is not in the document]
Comment (miguelpinto98) [reply src/app/app.module.ts:19]: To clarify my suggestion here. The aim here is to create feature/view modules (https://angular.io/guide/feature-modules). So, HomeComponent should be declared inside of HomeModule. An here, we should only use the HomeModule.

File: src/app/app-routing.module.ts [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+const routes: Routes = [
+	{
+		path: '',
+		pathMatch: 'full',
+		redirectTo: AppRoutingPath.Home
+	},
+	{
+		path: AppRoutingPath.Home,
+		pathMatch: 'full',
+		component: HomeComponent,
```
[the remark this answers is not in the document]
Comment (miguelpinto98) [reply src/app/app-routing.module.ts:19]: we also should change `component` to `loadChildren` to enable lazy-module.

For this case we should have something like this:
```
loadChildren: () => import('./pages/home.module').then(m => m.HomeModule)
```

**Review by miguelpinto98 — CHANGES_REQUESTED** [review]
Almost done 😄

File: src/app/app-routing.module.ts (hunk 13 above)
[the remark this answers is not in the document]
Comment (nelsonmestevao) [reply src/app/app-routing.module.ts]: Thank you for your suggestions, I believe it is as you suggested :blue_heart:

**Review by ricardogoncalves89 — APPROVED** [review]
🚀

**Pull request merged**